docs/migrating.rst

```diff
@@ -0,0 +1,225 @@
+longer available, the ``ndb.query.PropertyOrder`` class has been created to
+replace it.
+
+MessageProperty and EnumProperty
+================================
+
+These properties, from the ``ndb.msgprop`` module, depend on the Google
+Protocol RPC Library, or `protorpc`, which is not an `ndb` dependency. For
+this reason, they are not part of this version of `ndb`.
+
+Tasklets
+========
+
+When writing a `tasklet`, it is no longer necessary to raise a Return
+exception for returning the result. A normal return can be used instead::
+
+    @ndb.tasklet
+    def get_cart():
+        cart = yield CartItem.query().fetch_async()
+        return cart
+
+Note that "raise Return(cart)" can still be used, but it's not recommended.
+
+There are some methods from the ``tasklet`` module that are not implemented in
+this version of `ndb`, mainly because of changes in how an `ndb` context is
+created and used in this version:
+
+    - add_flow_exception.
+    - make_context.
+    - make_default_context.
+    - QueueFuture.
+    - ReducedFuture.
+    - SerialQueueFuture.
+    - set_context.
+    - toplevel.
+
+Utils
+=====
+
+The previous version of `ndb` included an ``ndb.utils`` module, which defined
+a number of methods that were mostly used internally. Some of those have been
+made obsolete by new Python 3 features, while others have been discarded due
+to implementation differences in the new `ndb`.
+
+Possibly the most used utility from this module outside of `ndb` code, is the
+``positional`` decorator, which declares that only the first `n` arguments of
+a function or method may be positional. Python 3 can do this using keyword-only
+arguments. What used to be written as::
+
+    @utils.positional(2)
+    def function1(arg1, arg2, arg3=None, arg4=None)
+        pass
+
+Will be written like this in the new version::
+
+    def function1(arg1, arg2, *, arg3=None, arg4=None)
+        pass
+
+Exceptions
+==========
+
+App Engine's legacy exceptions are no longer available, but `ndb` provides
+shims for most of them, which can be imported from the `ndb.exceptions`
+package, like this::
+
+    from ndb.exceptioms import BadRequestError, BadArgumentError
+
+Datastore API
+=============
+
+There are many differences bewteen the current Datastore API and the legacy App
+Engine Datastore. In most cases, where the public API was generally used, this
+should not be a problem. However, if you relied in your code on the private
+Datastore API, the code that does this will probably need to be rewritten.
+Specifically, any function or method that dealt directly with protocol buffers
+will no longer work. The Datastore `.protobuf` definitions have changed
+significantly from the public API used by App Engine to the current published
+API. Additionally, this version of NDB mostly delegates to
+`google.cloud.datastore` for parsing data returned by RPCs, which is a
+significant internal refactoring.
```

Comment:
I'm a little confused here. Are you saying that the old NDB library included some undocumented APIs that dealt directly with protocol buffers? Also, from reading the MIGRATION_NOTES, I'm not sure it would be possible for someone to tell which APIs dealt with protobufs. 

How about this:
The old NDB library included some undocumented APIs that dealt directly with Datastore protocol buffers. These APIs will no longer work. Rewrite any code that used the following classes, properties, or methods:
- ModelAdapter 
- Property._db_get_value, Property._db_set_value
- Property._db_set_compressed_meaning and Property._db_set_uncompressed_meaning 
- Model._deserialize and Model._serialize
- model.make_connection